src/main.go

```diff
@@ -0,0 +1,320 @@
+
+	parsedLensReviewJSON, err := json.Marshal(lensReviewMap)
+
+	if err != nil {
+		return fmt.Errorf("error marshaling lens review with workload id %s and lens alias %s: %v",
+			*lensReview.WorkloadId, *lensReview.LensReview.LensAlias, err)
+	}
+
+	parsedLensReviewJSON, err = addTypeToData(parsedLensReviewJSON)
+
+	if err != nil {
+		return err
+	}
+
+	wa.data = append(wa.data, parsedLensReviewJSON)
+
+	return nil
+}
+
+func (wa *WellArchitected) getLensReviewImprovements(workloadID *string, lensAlias *string) (*wellarchitected.ListLensReviewImprovementsOutput, error) {
+	lensReviewImprovementsInput := &wellarchitected.ListLensReviewImprovementsInput{
+		WorkloadId: workloadID,
+		LensAlias:  lensAlias,
+	}
+
+	lensReviewImprovementsOutput, err := wa.client.ListLensReviewImprovements(wa.ctx, lensReviewImprovementsInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get lens review improvements with workload id %s and lens alias %s: %v", *workloadID, *lensAlias, err)
+	}
+
+	return lensReviewImprovementsOutput, nil
+}
+
+func (wa *WellArchitected) parseLensReviewImprovements(
+	lensReviewImprovements *wellarchitected.ListLensReviewImprovementsOutput,
+	improvementSummary *types.ImprovementSummary,
+) error {
+	lensReviewImprovementsJSON, err := json.Marshal(lensReviewImprovements)
+
+	if err != nil {
+		return fmt.Errorf("error marshaling lens review improvements with workload id %s and lens alias %s: %v",
+			*lensReviewImprovements.WorkloadId, *lensReviewImprovements.LensAlias, err)
+	}
+
+	var lensReviewImprovementsMap map[string]interface{}
+	err = json.Unmarshal(lensReviewImprovementsJSON, &lensReviewImprovementsMap)
+
+	if err != nil {
+		return fmt.Errorf("error unmarshaling lens review improvements with workload id %s and lens alias %s: %v",
+			*lensReviewImprovements.WorkloadId, *lensReviewImprovements.LensAlias, err)
+	}
+
+	delete(lensReviewImprovementsMap, "ImprovementSummaries")
+	lensReviewImprovementsMap["ImprovementSummary"] = improvementSummary
+
+	parsedLensReviewJSON, err := json.Marshal(lensReviewImprovementsMap)
+
+	if err != nil {
+		return fmt.Errorf("error marshaling lens review improvements with workload id %s and lens alias %s: %v",
+			*lensReviewImprovements.WorkloadId, *lensReviewImprovements.LensAlias, err)
+	}
+
+	parsedLensReviewJSON, err = addTypeToData(parsedLensReviewJSON)
+
+	if err != nil {
+		return err
+	}
+
+	wa.data = append(wa.data, parsedLensReviewJSON)
+
+	return nil
+}
+
+func (wa *WellArchitected) sendDataToLogzio() error {
+	logzioSender, err := logzio.New(
+		os.Getenv("LOGZIO_TOKEN"),
+		logzio.SetUrl(os.Getenv("LOGZIO_URL")),
+		logzio.SetInMemoryQueue(true),
+		logzio.SetinMemoryCapacity(maxBulkSizeBytes),
+	)
+
+	if err != nil {
+		return fmt.Errorf("error creating LogzioSender: %v", err)
+	}
+
+	for _, data := range wa.data {
+		err := logzioSender.Send(data)
+
+		if err != nil {
+			return fmt.Errorf("error sending data to Logz.io: %v", err)
```

Comment:
Maybe add in a debug log the data that we couldn't send? It can give better indication on data that's being dropped in case we'll need to be bug detectives ;)